src/hotspot/os/aix/porting_aix.cpp

```diff
@@ -1035,7 +1035,7 @@ static bool search_file_in_LIBPATH(const char* path, struct stat64x* stat) {
 // specific AIX versions for ::dlopen() and ::dlclose(), which handles the struct g_handletable
 // This way we mimic dl handle equality for a library
 // opened a second time, as it is implemented on other platforms.
-void* Aix_dlopen(const char* filename, int Flags, const char** error_report) {
+void* Aix_dlopen(const char* filename, int *eno, int Flags, const char** error_report) {
```

Comment:
I think you also have to adapt the `Aix_dlopen()` declaration in the header file `porting_aix.hpp`

Reply:
Ok. I understood. The reason i was not getting error was i was compiling on Linux machine. I thought the jdk would throw errors atleast.

Reply:
Please switch Flags and eno
old: void* Aix_dlopen(const char* filename, int *eno, int Flags, const char** error_report) {
new: void* Aix_dlopen(const char* filename, int Flags, int *eno, const char** error_report) {

and I have to emphasize please also add this additional `int *eno` to the declaration of `Aix_dlopen()` in the `porting_aix.hpp` file